docs/tutorials/time-series.md

```diff
@@ -0,0 +1,766 @@
+    series, it cannot be deleted or modified.  Feldera implements an optimized version
+    of several operators for append-only collections.  In order to enable these
+    optimizations, the query engine needs to know which inputs are append-only.  The user
+    can provide this information to the engine using [`append_only` annotations](#append-only-tables).
+
+As a user, you do not need to understand the intricate details of time series
+processing in order to take advantage of these optimizations. Your simply
+communicate information about the input data to the query engine using
+annotations.  Feldera uses this information to apply optimizations automatically
+in a way that does not affect the output of the queries.  **Assuming that user
+annotations are correct, the optimized query plan produces the same outputs as
+the unoptimized plan.**
+
+**Users can control the timing of the output of time series queries**.  Users
+can further take advantage of the monotonicity of time series inputs to control
+**when** the output of a query is produced by Feldera using
+[`emit_final`](#emitting-final-values-of-a-view-with-emit_final)
+annotations.
+
+In the rest of this guide, we discuss these concepts in more detail.
+
+## Timestamp columns and lateness
+
+A timestamp column is a column in a SQL table or view that indicates a specific
+point in time associated with each row.
+
+  * There can be any number of timestamp columns in a table. For instance, a
+    table that describes taxi rides can store the time when the ride order was
+    received, pickup time, dropoff time, and the time when the payment was
+    processed.
+
+  * Timestamp columns can have any temporal (`TIMESTAMP`, `DATE`, `TIME`) or
+    numeric type.
+
+**A timestamp column can grow monotonically or almost monotonically**. Time
+series data often arrives ordered by timestamp, i.e., every event has the same
+or larger timestamp than the previous event.  In some cases, events can get
+reordered and delayed, but this delay is bounded, e.g., it may not exceed 1
+hour.  We refer to this bound as **lateness**.
+
+**Lateness** is a constant bound `L` associated with a timestamp column in a
+table or view, such that if the table contains a record with timestamp equal to
+`TS`, then any future records inserted to or removed from the table will have
+timestamps greater than or equal to `TS - L`.  In other words, updates to the
+table cannot arrive more than `L` time units out of order.
+
+The Feldera query engine uses lateness information to optimize the execution of
+queries.
+
+### Specifying lateness for tables and views
+
+To specify lateness for a table column, the column declaration is annotated with
+the `LATENESS` attribute:
+
+```sql
+CREATE TABLE purchase (
+   customer_id INT,
+   ts TIMESTAMP NOT NULL LATENESS INTERVAL 1 HOURS,
+   amount BIGINT
+);
+```
+
+The value of the `LATENESS` attribute is an expression that evaluates to a
```

Comment:
Might want to use more accessible language. Expression that evaluates to a constant value could instead just be "constant value" no?